src/interfaces.ts

```diff
@@ -38,13 +38,15 @@ export interface IFormMethods<T = any> {
     setProperty: (prop: keyof T, value: T[keyof T]) => any;
     setModel: (model: { [name in keyof T]?: any }) => any;
     setFieldToTouched: (prop: keyof T) => any;
+    handleSubmit: (submit: () => Promise<any>) => () => any;
```

Comment:
I suggest something like this for submit interface `submit: <T = any>(...args: any[]) => Promise<T>`

Reply:
actually, values can be injected by handleSubmit method.
But generic for the Promise is still a good thing.

Reply:
Updated to use a generic for the Promise and inject form values in handleSubmit.